### PR TITLE
Fix #4311

### LINKED ISSUE #4311: read_ch_connectivity fails when channel has no neighbours defined.
I computed a neighbours structure for an EGI 256 electrodes layout using fieldtrip and tried to read that mat file with `read_ch_connectivity`.

Electrode 31 in my montage has no neighbours, so the entire function failed, at two places.

1) `_recursive_flatten` assumes the `cell` parameter passed has at least one element.

This is a possible fix (just added the if condition):
``` Python
def _recursive_flatten(cell, dtype):
    """Unpack mat files in Python."""
    if len(cell) > 0:
        while not isinstance(cell[0], dtype):
            cell = [c for d in cell for c in d]
    return cell
```

2) `_ch_neighbor_connectivity` computes the set subtraction to raise an error:
`rest = set(ch_names) - set_neighbors` with the error "Some of your neighbors are not present in the list of channel names". In this case the error message and the operation do not match. The way I see it, it is an error only if the neighbours list has at least one electrode that it is not present in `ch_names`. If this is the point, then the operation should be `rest = set_neighbors - set(ch_names)`

### PR DESCRIPTION
**Pull request by fraimondo**
This should fix #4311

### CODE REVIEW

**Comment by agramfort** [comment]
can you add a test?

File: mne/channels/tests/test_channels.py [hunk 1]
```diff
@@ -161,6 +161,26 @@ def test_read_ch_connectivity():
...
+    mat = dict(neighbours=nbh)
+    mat_fname = op.join(tempdir, 'test_isolated_mat.mat')
+    savemat(mat_fname, mat, oned_as='row')
+    ch_connectivity, ch_names = read_ch_connectivity(mat_fname)
+    x = ch_connectivity.todense()
+    assert_equal(x.shape[0], len(ch_names))
+    assert_equal(x.shape, (4, 4))
+    assert_true(np.all(x.diagonal()))
+    assert_false(np.any(x[0, 1:]))
+    assert_false(np.any(x[1:, 0]))
```
Comment (Eric89GXL) [inline mne/channels/tests/test_channels.py]: can you add a test for the conditional you fixed (involving `set`)?

**Comment by fraimondo** [comment]
Done!

**Comment by agramfort** [comment]
thx @fraimondo

**Comment by fraimondo** [comment]
Always a pleasure

**Pull request merged**